examples/deadline/servermain/server.go

```diff
@@ -0,0 +1,62 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package main
+
+import (
+	"context"
+	"log"
+	"net"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc/examples/deadline/client"
+
+	"google.golang.org/grpc"
+	pb "google.golang.org/grpc/examples/helloworld/helloworld"
+)
+
+type server struct{}
+
+// SayHello implements helloworld.GreeterServer
+func (s *server) SayHello(ctx context.Context, in *pb.HelloRequest) (*pb.HelloReply, error) {
+	name := in.Name
+	if strings.HasPrefix(name, "[propagate me]") {
+		time.Sleep(800 * time.Millisecond)
+		name = strings.TrimPrefix(name, "[propagate me]")
+		return client.ConnectAndRequest(name)
```

Comment:
Clients should not be connecting in the RPC path.  The server should create a client (to itself) when it starts up.